cache/manager.go

```diff
@@ -0,0 +1,42 @@
+package cache
+
+import (
+	"sync"
+	"time"
+)
+
+type cacheManager struct {
+	avaliablesCache map[string]CacheClient
+}
+
+var instance CacheManager = nil
+
+func NewCacheManager() CacheManager {
```

Comment:
Really nice cache implementation, I think it can be a really nice improvement. 
I'm just wondering where/how exactly is this being used/invoked. Maybe I'm missing how this gets injected in the module.